### PR TITLE
[3.8] bpo-36384: Leading zeros in IPv4 addresses are no longer tolerated (GH-25099)

### PR DESCRIPTION
**Pull request by achraf-mer**
https://bugs.python.org/issue36384

### CODE REVIEW

**Comment by the-knights-who-say-ni** [comment]
Hello, and thanks for your contribution!

I'm a bot set up to make sure that the project can legally accept this contribution by verifying everyone involved has signed the [PSF contributor agreement](https://www.python.org/psf/contrib/contrib-form/) (CLA).

## Recognized GitHub username

We couldn't find a [bugs.python.org](https://bugs.python.org/) (b.p.o) account corresponding to the following GitHub usernames:

@achraf-mer

This might be simply due to a missing "GitHub Name" entry in one's b.p.o account settings. This is necessary for legal reasons before we can look at this contribution. Please follow [the steps outlined in the CPython devguide](https://devguide.python.org/pullrequest/#licensing) to rectify this issue.




You can [check yourself](https://check-python-cla.herokuapp.com/) to see if the CLA has been received.

Thanks again for the contribution, we look forward to reviewing it!

**Comment by ambv** [comment]
See: https://bugs.python.org/issue36384#msg392684

> Due to the relative obscurity of the bug and potential disruption of the fix, I decided not to include it in 3.8.

**Comment by achraf-mer** [comment]
> See: https://bugs.python.org/issue36384#msg392684
> 
> > Due to the relative obscurity of the bug and potential disruption of the fix, I decided not to include it in 3.8.

@ambv please take note of my reply here, on why I think this is needed still: https://bugs.python.org/issue36384#msg399801 thx

File: Doc/library/ipaddress.rst [hunk 1]
```diff
@@ -117,6 +116,12 @@ write code that handles both IP versions correctly.  Address objects are
    >>> ipaddress.IPv4Address(b'\xC0\xA8\x00\x01')
    IPv4Address('192.168.0.1')
 
+   .. versionchanged:: 3.8
+
+      Leading zeros are no longer tolerated and are treated as an error.
+      IPv4 address strings are now parsed as strict as glibc
+      :func:`~socket.inet_pton`.
+
```
Comment (ambv) [inline Doc/library/ipaddress.rst]: This is incorrect. You should use the wording from GH-25099 for `.. versionchanged:: 3.8` *and* then add `.. versionchanged:: 3.8.12` with the content you have here.

**Review by ambv — CHANGES_REQUESTED** [review]
Thank you for the backport. We need two more things. The first is mentioned in an inline comment. And the second is about "What's New". See how GH-25099 also touches `Doc/whatsnew/3.9.rst`? This backport will need to do the same for `Doc/whatsnew/3.8.rst`, adding "Notable Changes in Python 3.8.12". You can copy the text verbatim from the respective whatsnew edit in GH-25099.

File: Doc/library/ipaddress.rst (hunk 1 above)
Comment (achraf-mer) [reply Doc/library/ipaddress.rst]: Done, PTAL.

**Comment by achraf-mer** [comment]
I have made the requested changes; please review again.

**Review by ambv — APPROVED** [review]

**Comment by achraf-mer** [comment]
@ambv I am pretty new with this workflow, would you mind helping me understand what the next step is, or perhaps the timeline of when this PR can get merged, and how it would be possible for us to start using the new python version, (let's say from conda since that's what we use).
Thanks

**Comment by ambv** [comment]
@achraf-mer, assuming all tests pass on this PR, I will merge it. It will be released as Python 3.8.12 on August 30th alongside Python 3.9.7. I will also make pull requests to the 3.9, 3.10, and main (3.11) branches to amend the `.. versionchanged:: 3.8.12` information you provide here in the docs of those respective branches, to make this information easier to find.

This PR is currently the only fix slated for 3.8.12 (there are literally only 3 other commits on the branch, two are doc updates and one is a test improvement).

Not sure how soon conda will release 3.8.12. Looking at 3.8.11, it took them 37 days from release (2021-06-28) to availability on https://anaconda.org/anaconda/python/files (2021-08-04).

**Comment by ambv** [comment]
@achraf-mer, thank you for taking your time reporting and contributing the backport. You likely saved us a lot of grief we'd otherwise face due to the overblown severity mark on the CVE.

**Comment by achraf-mer** [comment]
> @achraf-mer, thank you for taking your time reporting and contributing the backport. You likely saved us a lot of grief we'd otherwise face due to the overblown severity mark on the CVE.

@ambv :+1:   thanks for the prompt reply.

**Pull request merged**